client/src/pages/Discovery/Discovery.tsx

```diff
@@ -30,14 +30,18 @@ export default function Discovery(): JSX.Element {
   const history = useHistory();
 
   useEffect(() => {
-    getAllContests().then((data) => {
-      if (data.error) {
-        updateSnackBarMessage(data.error.message);
-      } else {
-        setContestObj(data.contest);
-      }
-    });
-  }, [updateSnackBarMessage]);
+    const fetchAllcontests = () => {
+      history.push(`/discovery`);
```

Comment:
Why are we pushing the '/discovery' route onto the history stack as this useEffect will only run when the user is on the discovery page?